strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/maven/MavenArtifactControllerTest.java

```diff
@@ -841,15 +841,39 @@ public void testThatMavenProxyRepoDoesNotBlockTheConnectionPool()
                .get(url)
                .peek()
                .then()
-               .statusCode(HttpStatus.NOT_FOUND.value());
+               .statusCode(HttpStatus.OK.value());
+    }
+
+    @Test
+    public void shouldDownloadProxiedSnapshotArtifact()
+            throws Exception
+    {
+        String url = getContextBaseUrl() +
```

Comment:
Generate a snapshot Maven artifact in snapshot repository (see how it's done in other tests). Then include this repository in a group repository (if it's not already included) and just resolve it from there, instead of relying on an outside service. This way you'll always have the artifact.